Sources/StreamChatSwiftUI/ChatChannel/Reactions/MessageActions/DefaultMessageActions.swift

```diff
@@ -110,6 +110,36 @@ extension MessageAction {
             messageActions.append(copyAction)
         }
 
+        if message.isRootOfThread {
+            let messageController = InjectedValues[\.utils]
+                .channelControllerFactory
+                .makeMessageController(for: message.id, channelId: channel.cid)
+            // At the moment, this is the only way to know if we are inside a thread.
+            // This should be optimised in the future and provide the view context.
+            let isInsideThreadView = messageController.replies.count > 0
+            if isInsideThreadView {
+                let markThreadUnreadAction = markThreadAsUnreadAction(
+                    messageController: messageController,
+                    message: message,
+                    onFinish: onFinish,
+                    onError: onError
+                )
+                messageActions.append(markThreadUnreadAction)
+            }
```

Comment:
This is kinda hacky, but it was the only way I found to know if the messages actions are inside a thread view or not. @martinmitrevski Let me know if there is any better way at the moment